src/libraries/System.IO.FileSystem/tests/FileStream/WriteAsync.cs

```diff
@@ -220,7 +220,7 @@ public Task ManyConcurrentWriteAsyncs()
                 {
                     writes[i] = WriteAsync(fs, expectedData, i * writeSize, writeSize, cancellationToken);
                     Assert.Null(writes[i].Exception);
-                    if (useAsync)
+                    if (useAsync && PlatformDetection.IsLegacyFileStreamEnabled)
```

Comment:
Would you mind adding a comment to explain why this case will only work for the legacy version?